cura/PrinterOutput/PrinterOutputController.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2017 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from UM.Logger import Logger
+
+MYPY = False
+if MYPY:
+    from cura.PrinterOutput.PrintJobOutputModel import PrintJobOutputModel
+    from cura.PrinterOutput.ExtruderOuputModel import ExtruderOuputModel
+    from cura.PrinterOutput.PrinterOutputModel import PrinterOutputModel
+
+
+class PrinterOutputController:
```

Comment:
I'm trying to figure out the value of the `PrinterOutputController` and its relation to the `PrinterOutputModel`. Is the controller needed? Right now it just looks like the controller contains the verbs and the model contains the nouns, or more accurately, the data is in the model and the functions which operate on the data live in the controller. In a way OOP was about combining data with the functions that operate on it. i.e. objects have data and can do things.

Reply:
Except that in this case, it would cause a slew of uneeded extra objects; In order for it to work you'd have a ClusterPrintjobOutputModel, ClusterExtruderOutputModel, LegacuUM3ExtruderOutputModel, etc. This way, i can keep the object itself as a proxy, which delegates the changing of a parameter to the controller (that is specific to the type of printer it's actually connected with)